From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 5 Sep 2021 16:13:03 +0200
Subject: [PATCH] {BUG in 5.15-rc1} drivers/gpu/drm/i915/gem/selftests: do not
 use an uninitialized variable

clang on x86-64 reports:

    drivers/gpu/drm/i915/gem/selftests/i915_gem_dmabuf.c:63:6: error:
    variable 'import_obj' is used uninitialized whenever 'if' condition
    is true [-Werror,-Wsometimes-uninitialized]
            if (import != &obj->base) {
                ^~~~~~~~~~~~~~~~~~~~
    drivers/gpu/drm/i915/gem/selftests/i915_gem_dmabuf.c:80:22: note
    uninitialized use occurs here
            i915_gem_object_put(import_obj);
                                ^~~~~~~~~~
    drivers/gpu/drm/i915/gem/selftests/i915_gem_dmabuf.c:63:2: note:
    remove the 'if' if its condition is always false
            if (import != &obj->base) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/gpu/drm/i915/gem/selftests/i915_gem_dmabuf.c:38:46: note:
    initialize the variable 'import_obj' to silence this warning
            struct drm_i915_gem_object *obj, *import_obj;
                                                        ^
                                                         = NULL
---
 drivers/gpu/drm/i915/gem/selftests/i915_gem_dmabuf.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/gem/selftests/i915_gem_dmabuf.c b/drivers/gpu/drm/i915/gem/selftests/i915_gem_dmabuf.c
index ffae7df5e4d7..e655e8eca372 100644
--- a/drivers/gpu/drm/i915/gem/selftests/i915_gem_dmabuf.c
+++ b/drivers/gpu/drm/i915/gem/selftests/i915_gem_dmabuf.c
@@ -63,7 +63,7 @@ static int igt_dmabuf_import_self(void *arg)
 	if (import != &obj->base) {
 		pr_err("i915_gem_prime_import created a new object!\n");
 		err = -EINVAL;
-		goto out_import;
+		goto out_dmabuf /* BUG */;
 	}
 	import_obj = to_intel_bo(import);
 
@@ -128,7 +128,7 @@ static int igt_dmabuf_import_same_driver_lmem(void *arg)
 		pr_err("i915_gem_prime_import failed with the wrong err=%ld\n",
 		       PTR_ERR(import));
 		err = PTR_ERR(import);
-	}
+	} else { err = 0; } /* BUG */
 
 	dma_buf_put(dmabuf);
 out:
@@ -180,7 +180,7 @@ static int igt_dmabuf_import_same_driver(struct drm_i915_private *i915,
 	if (import == &obj->base) {
 		pr_err("i915_gem_prime_import reused gem object!\n");
 		err = -EINVAL;
-		goto out_import;
+		goto out_dmabuf /* BUG */;
 	}
 
 	import_obj = to_intel_bo(import);
-- 
